### PR TITLE
Add support for "Having" filter in druid

### PR DESCRIPTION
**Pull request by x4base**

### CODE REVIEW

File: caravel/models.py [hunk 1]
```diff
@@ -1265,6 +1243,59 @@ def recursive_get_fields(_conf):
             query=query_str,
             duration=datetime.now() - qry_start_dttm)
 
+    @staticmethod
+    def get_filters(filter):
```
Comment (mistercrunch) [inline caravel/models.py]: `filter` is a standard lib, main namespace function. Here you are redefining it for the scope of this function. Use `filter_` or some other name instead.

**Comment by mistercrunch** [comment]
This is great, I finally took the time to test it. The last missing thing for me is to make sure that only metrics are shown in the Having dropdown field.

**Comment by x4base** [comment]
Do you mean the options should not include dimensions? But sometimes the user may want to filter the dimensions in the post-aggregation stage, right?

**Comment by mistercrunch** [comment]
Oh, if that works then it's good!

**Comment by mistercrunch** [comment]
Actually it fails if you try to to use `==` on a dimension using a string...

**Comment by x4base** [comment]
Sorry I didn't test the filter with strings, it should work now

**Comment by mistercrunch** [comment]
PR needs a rebase ...

**Comment by x4base** [comment]
Updated

**Pull request merged**